ArchiSteamFarm/Steam/Bot.cs

```diff
@@ -3471,6 +3326,96 @@ public sealed class Bot : IAsyncDisposable {
 			}
 		}
 
+		private async Task<bool> SendMessagePart(ulong steamID, string messagePart) {
+			if ((steamID == 0) || !new SteamID(steamID).IsIndividualAccount) {
+				throw new ArgumentOutOfRangeException(nameof(steamID));
+			}
+
+			if (string.IsNullOrEmpty(messagePart)) {
+				throw new ArgumentNullException(nameof(messagePart));
+			}
+
+			if (!IsConnectedAndLoggedOn) {
+				return false;
+			}
+
+			await MessagingSemaphore.WaitAsync().ConfigureAwait(false);
+
+			try {
+				for (byte i = 0; (i < WebBrowser.MaxTries) && IsConnectedAndLoggedOn; i++) {
+					EResult result = await ArchiHandler.SendMessage(steamID, messagePart).ConfigureAwait(false);
+
+					switch (result) {
+						case EResult.Busy:
+						case EResult.Fail:
+						case EResult.RateLimitExceeded:
+						case EResult.ServiceUnavailable:
+						case EResult.Timeout:
+							await Task.Delay(5000).ConfigureAwait(false);
+
+							continue;
+						case EResult.OK:
+							return true;
+						default:
+							ArchiLogger.LogGenericError(string.Format(CultureInfo.CurrentCulture, Strings.WarningUnknownValuePleaseReport, nameof(result), result));
+
+							return false;
+					}
+				}
+
+				return false;
+			} finally {
+				MessagingSemaphore.Release();
+			}
+		}
+
+		private async Task<bool> SendMessagePart(ulong chatGroupID, ulong chatID, string messagePart) {
+			if (chatGroupID == 0) {
+				throw new ArgumentOutOfRangeException(nameof(chatGroupID));
+			}
+
+			if (chatID == 0) {
+				throw new ArgumentOutOfRangeException(nameof(chatID));
+			}
+
+			if (string.IsNullOrEmpty(messagePart)) {
+				throw new ArgumentNullException(nameof(messagePart));
+			}
+
+			if (!IsConnectedAndLoggedOn) {
+				return false;
+			}
+
+			await MessagingSemaphore.WaitAsync().ConfigureAwait(false);
+
+			try {
+				for (byte i = 0; (i < WebBrowser.MaxTries) && IsConnectedAndLoggedOn; i++) {
+					EResult result = await ArchiHandler.SendMessage(chatGroupID, chatID, messagePart).ConfigureAwait(false);
+
+					switch (result) {
+						case EResult.Busy:
+						case EResult.Fail:
+						case EResult.RateLimitExceeded:
+						case EResult.ServiceUnavailable:
+						case EResult.Timeout:
+							await Task.Delay(5000).ConfigureAwait(false);
+
+							continue;
+						case EResult.OK:
+							return true;
+						default:
+							ArchiLogger.LogGenericError(string.Format(CultureInfo.CurrentCulture, Strings.WarningUnknownValuePleaseReport, nameof(result), result));
+
+							return false;
+					}
+				}
+
+				return false;
+			} finally {
+				MessagingSemaphore.Release();
+			}
+		}
```

Comment:
We can de-duplicate this code by using a delegate. Of course this would require us to check `steamId`, `chatGroupID`, `chatID` and `messagePart` for illegal values in other places, which I have not done with this code edit (before lines 775 and 808).
```suggestion
		private async Task<bool> SendMessagePart(Func<Task<EResult>> sendingDelegate) {
			if (!IsConnectedAndLoggedOn) {
				return false;
			}

			await MessagingSemaphore.WaitAsync().ConfigureAwait(false);

			try {
				for (byte i = 0; (i < WebBrowser.MaxTries) && IsConnectedAndLoggedOn; i++) {
					EResult result = await sendingDelegate().ConfigureAwait(false);

					switch (result) {
						case EResult.Busy:
						case EResult.Fail:
						case EResult.RateLimitExceeded:
						case EResult.ServiceUnavailable:
						case EResult.Timeout:
							await Task.Delay(5000).ConfigureAwait(false);

							continue;
						case EResult.OK:
							return true;
						default:
							ArchiLogger.LogGenericError(string.Format(CultureInfo.CurrentCulture, Strings.WarningUnknownValuePleaseReport, nameof(result), result));

							return false;
					}
				}

				return false;
			} finally {
				MessagingSemaphore.Release();
			}
		}
```